### PR TITLE
Fixes #14357: Add a method that allows reporting lines from a file

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/14357

### CODE REVIEW

**Comment by amousset** [comment]
I will add a reporting test too.

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

File: tree/30_generic_methods/file_report_content.cf [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name File report content
+# @description Report the content of a file
+# @documentation Reports the content of a file.
```
Comment (ncharles) [inline tree/30_generic_methods/file_report_content.cf]: can you choose between Report and Reports ?

File: tree/30_generic_methods/file_report_content.cf [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+# @parameter_constraint context "regex" : "^\\d*$"
+#
+# @class_prefix file_report_content
+# @class_parameter target
+
+bundle agent file_report_content(target, regex, context)
+{
+  vars:
+      "old_class_prefix"  string => canonify("file_report_content_${target}");
+      "args"               slist => { "${target}", "${regex}", "${context}" };
```
Comment (ncharles) [inline tree/30_generic_methods/file_report_content.cf]: this should be in !pass1 is we don't want to rewrite the class prefix based on default (like in directory_absent)

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Pull request merged**